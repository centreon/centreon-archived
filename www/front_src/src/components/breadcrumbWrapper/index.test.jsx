import React from 'react';
import configureStore from 'redux-mock-store';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import BreadcrumbWrapper from '.';

const mockStore = configureStore([]);

describe('BreadcrumbWrapper', () => {
  it('renders', () => {
    // Initialize mockstore with empty state
    const initialState = {
      navigation: {
        items: [
          {
            page: '1',
            label: 'Home',
            is_react: false,
            url: './include/home/home.php',
            options: null,
            children: [
              {
                groups: [],
                page: '103',
                label: 'Custom Views',
                is_react: true,
                url: '/home/customViews',
                options: null,
              },
            ],
          },
        ],
      },
    };
    const store = mockStore(initialState);

    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <BreadcrumbWrapper path='/home/customViews'>
            <div>My Page</div>
          </BreadcrumbWrapper>
        </MemoryRouter>
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
