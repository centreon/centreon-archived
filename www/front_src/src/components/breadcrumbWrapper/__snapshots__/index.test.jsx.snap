// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbWrapper renders 1`] = `
.emotion-2 {
  padding: 4px 16px;
}

.emotion-2 .MuiBreadcrumbs-li {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 {
  font-size: 12px;
}

<nav
  aria-label="Breadcrumb"
  class="MuiTypography-root MuiBreadcrumbs-root emotion-2 MuiTypography-body1 MuiTypography-colorTextSecondary"
>
  <ol
    class="MuiBreadcrumbs-ol"
  >
    <li
      class="MuiBreadcrumbs-li"
    >
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover emotion-0 MuiTypography-colorInherit"
        href="/main.php?p=1"
      >
        Home
      </a>
    </li>
    <li
      aria-hidden="true"
      class="PrivateBreadcrumbSeparator-root-41 MuiBreadcrumbs-separator"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        />
      </svg>
    </li>
    <li
      class="MuiBreadcrumbs-li"
    >
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover emotion-0 MuiTypography-colorTextPrimary"
        href="/home/customViews"
      >
        Custom Views
      </a>
    </li>
  </ol>
</nav>
`;
